### PR TITLE
Add support for multiple controllers

### LINKED ISSUE #18: Add Voltex FW mode
Been doing some thinking about this one, and it looks like we are going to need a special FW mode for voltex. There are a couple of hurdles:

1. The [encoders used in cryto's voltex controller](https://www.sensatec.co.jp/e/products/catalog.php/293/ST-ELV-24X35B-K_EN.pdf]) use a single 3 pin connector with a variable voltage output. Our design was made with the assumption that controllers will always use encoders with 2 data lines (either in a 2x 3pin format or combined in a single 4pin). As this is not the case, we can re-use one of the 3-pin encoder inputs as an input for these voltex encoders, but we will have to change how the input is handled in FW.
2. The lighting effects run down the sides on the SVRE9, which would also need special FW. Not even sure what type of LEDs this thing uses, but I will experiment and find out.

For now I think it is acceptable to have this a compile-time flag, eventually I would like this to be toggled using a button combo.

Assigning this to myself for now, @supervaka feel free to chime in with your thoughts

### PR DESCRIPTION
**Pull request by ASleepyCat**
Fairly large PR. Closes #18 and #75.

This PR adds basic support for SOUND VOLTEX controllers, and refactors the code base to be structured for multi-device support.
Tested on my SVSE5 and PHOENIXWAN LMT controllers.

* Add support for switching between device types on boot-up
  * Simple one-time combo, set once and it'll remember what controller type it's supposed to be
* Add support for SOUND VOLTEX controllers
  * Add support for AC-style knobs
    * Seems like the way AVRs do ADC doesn't result in crazy jittering like it does on arcin. Thus, no filtering needed on our end.
    * I haven't added any sensitivity options since it feels about right anyway
  * Add HID support for SOUND VOLTEX controllers
  * Add support for EAC for SOUND VOLTEX controllers
  * I haven't tested the stock knobs that are on the FAUCETWO, but I assume they'll work when wired up to the TT input headers
  * By default, AC-style knobs are enabled. In the future, there should be a combo to enable switching between knob types.
* Flip TT sensitivity and deadzone combos
  * I think this is more accurate to the stock board
* Add section in README on how to correctly wire up the board for specific controller types
* Add support for second QE sensor
* Misc. refactoring
  * Set default `BEEF_LED_REFRESH` value to 240
  * Refactor `Bpm` to return updated value on `update()`, only run `Bpm` code when relevant Bar LED mode is enabled
  * Replace some `#define` consts with `enum`

### CODE REVIEW

File: fw/devices/iidx/iidx_combo.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#include "iidx_combo.h"
+#include "iidx_usb.h"
+
+namespace Iidx {
```
Comment (HWXLR8) [inline fw/devices/iidx/iidx_combo.cpp:4]: `Iidx` hurts my brain to read, thoughts on using all caps?

File: fw/devices/sdvx/sdvx_combo.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include "../beef.h"
+#include "sdvx_combo.h"
+
+namespace Sdvx {
```
Comment (HWXLR8) [inline fw/devices/sdvx/sdvx_combo.cpp:4]: Same comment as above, all caps would be nice.

File: fw/makefile [hunk 3]
```diff
@@ -5,11 +5,11 @@ F_CPU = 16000000
 F_USB = $(F_CPU)
 OPTIMIZATION = 2
 TARGET = beef
-BEEF_LED_REFRESH ?= 40
+BEEF_LED_REFRESH ?= 240
```
Comment (HWXLR8) [inline fw/makefile:8]: reason for this change?

**Review by HWXLR8 — CHANGES_REQUESTED** [review]
Absolutely fantastic work. I cannot thank you enough for your contributions to this project. Only minor concerns, will test on my board shortly.

File: fw/makefile (hunk 3 above)
Comment (ASleepyCat) [reply fw/makefile:8]: We have the headroom and I was getting tired of having to set a higher refresh rate, so I increased the default value.

File: fw/devices/iidx/iidx_combo.cpp (hunk 1 above)
Comment (ASleepyCat) [reply fw/devices/iidx/iidx_combo.cpp]: Fair enough, changed.

**Comment by ASleepyCat** [comment]
The only real issue I have with my changes is that I have to flip the order of the X/Y axis in the HID report. Not sure why it doesn't work in the proper order.

**Review by HWXLR8 — APPROVED** [review]
I have not been able to fully test this on my end, but I will approve as-is. I trust that you have done your due-diligence. Will test in the coming days.

**Pull request merged**